test/Microsoft.AspNetCore.Mvc.IntegrationTests/BinderTypeBasedModelBinderIntegrationTest.cs

```diff
@@ -234,8 +234,6 @@ private class Address
 
             // ModelState
             Assert.True(modelState.IsValid);
-
-            Assert.Equal(1, modelState.Keys.Count);
```

Comment:
Make your `Assert.Single(modelState.Keys)` change here. Otherwise are no longer checking what this assertion did.